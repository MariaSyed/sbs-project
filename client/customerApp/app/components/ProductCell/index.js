import React from 'react';
import {
  Image,
  StyleSheet,
  Text,
  TouchableWithoutFeedback,
  View,
} from 'react-native';

import { defaultPaddings } from '../../config';
import IconButton from '../IconButton';

export default ({
  productId,
  title,
  subtitle,
  price,
  imageSource,
  stores = ['-'],
  onAddToCart,
  onOpenDetails,
  ...props
}) => (
  <View style={styles.container}>
    <TouchableWithoutFeedback onPress={() => onOpenDetails(productId)}>
      <View style={styles.imageContainer}>
        <Image style={styles.productImage} source={imageSource} />
      </View>
    </TouchableWithoutFeedback>
    <TouchableWithoutFeedback onPress={() => onOpenDetails(productId)}>
      <View style={styles.body}>
        <Text style={styles.descriptionTitle}>{title}</Text>
        <Text style={styles.descriptionSubtitle}>{subtitle}</Text>
      </View>
    </TouchableWithoutFeedback>
    <View style={styles.actionsContainer}>
      <IconButton
        iconName="add-shopping-cart"
        label={price}
        onPress={() => onAddToCart(productId)}
      />
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingVertical: 10,
    backgroundColor: '#fff',
  },
  imageContainer: {
    width: 100,
    justifyContent: 'center',
    paddingHorizontal: defaultPaddings.paddingHorizontal,
  },
  productImage: {
    width: 100 - 2 * defaultPaddings.paddingHorizontal,
    height: 80,
  },
  body: {
    flex: 1,
    paddingTop: 10,
    paddingHorizontal: 5,
  },
  descriptionTitle: {
    fontSize: 17,
    fontWeight: '600',
    marginBottom: 5,
  },
  descriptionSubtitle: {
    color: '#808080',
    fontSize: 14,
  },
  actionsContainer: {
    width: 75,
    paddingRight: defaultPaddings.paddingHorizontal,
    justifyContent: 'center',
  },
});
