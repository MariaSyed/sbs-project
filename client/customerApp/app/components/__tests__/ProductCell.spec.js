/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ProductCell from '../ProductCell';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ProductCell
        productId={1}
        price="2.50"
        title="Title"
        stores={['Lidl']}
        imageSource={{
          uri: 'https://facebook.github.io/react/img/logo_og.png',
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
