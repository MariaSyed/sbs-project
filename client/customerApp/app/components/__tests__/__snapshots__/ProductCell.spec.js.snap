// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "paddingVertical": 10,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "justifyContent": "center",
        "paddingHorizontal": 15,
        "width": 100,
      }
    }
    testID={undefined}
  >
    <Image
      source={
        Object {
          "uri": "https://facebook.github.io/react/img/logo_og.png",
        }
      }
      style={
        Object {
          "height": 70,
          "width": 60,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 5,
      }
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 17,
          "fontWeight": "600",
          "marginBottom": 5,
        }
      }
    >
      Title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#808080",
          "fontSize": 14,
        }
      }
    >
      Lidl
    </Text>
  </View>
  <View
    style={
      Object {
        "justifyContent": "center",
        "paddingRight": 15,
        "width": 75,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(230,37,101)",
            "borderColor": "rgb(230,37,101)",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
            "paddingHorizontal": 5,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 20,
              },
              Object {
                "color": "#fff",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 13,
              "fontWeight": "bold",
              "marginTop": 5,
            }
          }
        >
          2.50 €
        </Text>
      </View>
    </View>
  </View>
</View>
`;
